Functions.Templates/Templates/AuthenticationEventsTrigger-CSharp/AuthenticationEventsTriggerCSharp.cs

```diff
@@ -0,0 +1,49 @@
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents;
+using Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents.Framework;
+using Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents.TokenIssuanceStart;
+using Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents.TokenIssuanceStart.Actions;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+
+namespace Company.Function
+{
+    /// <summary>Example functions for token augmentation</summary>
+    public static class AuthenticationEventFunctions
+    {
+        /// <summary>The entry point for the Azure Function</summary>
+        /// <param name="request">Strongly Typed request data for a token issuance start request</param>
+        /// <param name="log">Logger</param>
+        /// <returns>The augmented token response or error.</returns>
+        [FunctionName("onTokenIssuanceStart")]
```

Comment:
Function name should also be `AuthenticationEventsTriggerCSharp`

Reply:
Not sure about this one, is it required as this function is for the onTokenIssuanceStart event, later we will have more functions in this file. (C# can have multiple functions in a single class)